File: app/src/main/java/com/photoeditorsdk/android/app/MainActivity.java
package com.photoeditorsdk.android.app;


import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;

public class MainActivity extends Activity implements View.OnClickListener {

    private MainActivity context;
    private View createBtn;
    private View playBtn;
    private View editBtn;
    private View settingsBtn;

    //선택한 버튼이 어떤 메뉴를 선택했는지 확인하기 위한 변수
    //create와 play에서 선택한 이미지가 캐시디렉토리에 저장되어 올바른 액티비티로 넘어가게 하기 위함
    public int menuflag = 0;
    //각 메뉴 번호
    public static final int ACTION_REQUEST_CREATE = 6;
    public static final int ACTION_REQUEST_PLAY = 7;
    public static final int ACTION_REQUEST_EDIT = 8;
    public static final int ACTION_REQUEST_SETTINGS = 9;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initView(); //뷰 초기화
    }

    //뷰 초기화
    private void initView() {
        context = this;

        //메인메뉴의 각 버튼을 연결
        createBtn = (ImageButton)findViewById(R.id.create_btn);
        playBtn = (ImageButton)findViewById(R.id.play_btn);
        editBtn = (ImageButton)findViewById(R.id.edit_btn);
        settingsBtn = (ImageButton)findViewById(R.id.settings_btn);

        //클릭 리스너 설정
        createBtn.setOnClickListener(this);
        playBtn.setOnClickListener(this);
        editBtn.setOnClickListener(this);
        settingsBtn.setOnClickListener(this);
    }

    //버튼을 클릭하면 크리에이트, 플레이는 갤러리에서 사진을 선택할 수 있게 됨
    //Edit은 따로 설정 필요
    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.create_btn:
                menuflag = ACTION_REQUEST_CREATE;   //인텐트를 넘길 고유번호 설정
                selectMenu();                  //인텐트 넘기기
                break;
            case R.id.play_btn:
                menuflag = ACTION_REQUEST_PLAY;     //인텐트를 넘길 고유번호 설정
                selectMenu();                  //인텐트 넘기기
                break;
            case R.id.edit_btn:
                menuflag = ACTION_REQUEST_EDIT;
                selectMenu();                       //인텐트 설정
                break;
            case R.id.settings_btn:
                menuflag = ACTION_REQUEST_SETTINGS;
                selectMenu();                       //인텐트 설정
                break;
        }
    }

    //클릭한 버튼이 어떤 메뉴인지 확인하고 해당 액티비티로 넘어가게 함
    private void selectMenu() {
        switch (menuflag) {
            case ACTION_REQUEST_CREATE:
                Intent intent1 = new Intent(getApplicationContext(), CallGallery.class);
                intent1.putExtra("createCode", ACTION_REQUEST_CREATE);
                startActivity(intent1);
                break;
            case ACTION_REQUEST_PLAY:
                //갤러리 오픈으로 넘어감
                Intent intent2 = new Intent(getApplicationContext(), CallGallery.class);
                intent2.putExtra("playCode", ACTION_REQUEST_PLAY);
                startActivity(intent2);
                break;
            case ACTION_REQUEST_EDIT:
                //편집화면으로 넘어감
                Intent intent3 = new Intent(getApplicationContext(), EditSystem.class);
                startActivity(intent3);
                break;
            case ACTION_REQUEST_SETTINGS:
                //셋팅화면으로 넘어감
                Intent intent4 = new Intent(getApplicationContext(), SettingsSystem.class);
                startActivity(intent4);
                break;
        }
    }
}

File: app/src/main/java/com/photoeditorsdk/android/app/MainActivity.kt
package com.photoeditorsdk.android.app

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Intent
import android.media.Image
import android.net.Uri
import android.os.Bundle
import android.os.Environment
import android.provider.MediaStore
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.ImageButton
import android.widget.Toast
import ly.img.android.pesdk.PhotoEditorSettingsList
import ly.img.android.pesdk.assets.filter.basic.FilterPackBasic
import ly.img.android.pesdk.assets.font.basic.FontPackBasic
import ly.img.android.pesdk.assets.frame.basic.FramePackBasic
import ly.img.android.pesdk.assets.overlay.basic.OverlayPackBasic
import ly.img.android.pesdk.assets.sticker.emoticons.StickerPackEmoticons
import ly.img.android.pesdk.assets.sticker.shapes.StickerPackShapes
import ly.img.android.pesdk.backend.model.EditorSDKResult
import ly.img.android.pesdk.backend.model.constant.Directory
import ly.img.android.pesdk.backend.model.state.LoadSettings
import ly.img.android.pesdk.backend.model.state.SaveSettings
import ly.img.android.pesdk.ui.activity.PhotoEditorBuilder
import ly.img.android.pesdk.ui.model.state.*
import ly.img.android.pesdk.ui.panels.item.PersonalStickerAddItem
import ly.img.android.pesdk.ui.utils.PermissionRequest
import ly.img.android.serializer._3.IMGLYFileWriter
import java.io.File
import java.io.IOException

class MainActivity : Activity(), View.OnClickListener {
    var menuflag = 0;

    companion object {
        const val PESDK_RESULT = 1
        /*const val EDIT_RESULT = 2*/
        const val CREATE_RESULT = 3
        const val PLAY_RESULT = 4
        const val SETTINGS_RESULT = 5
    }

    @SuppressLint("WrongViewCast")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initView();
    }

    @SuppressLint("WrongViewCast")
    private fun initView(){

    /*activity_main 버튼 연결*/

        val create = findViewById<ImageButton>(R.id.create_btn)
        val play = findViewById<ImageButton>(R.id.play_btn)
        /*val edit = findViewById<ImageButton>(R.id.edit_btn)*/
        val settings = findViewById<ImageButton>(R.id.settings_btn)

        create.setOnClickListener(this)
        play.setOnClickListener(this)
        /*edit.setOnClickListener(this)*/
        settings.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when (v?.id){
            R.id.create_btn -> {
                menuflag = CREATE_RESULT
                selectMenu()}
            R.id.play_btn -> {
                menuflag = PLAY_RESULT
                selectMenu()}
            /*R.id.edit_btn -> {
                menuflag = EDIT_RESULT
                selectMenu()}*/
            R.id.settings_btn -> {
                menuflag = SETTINGS_RESULT
                selectMenu()}
        }
    }

    fun selectMenu(){
        when (menuflag){
            CREATE_RESULT -> {
            val intent1 = Intent(this, CallGallery::class.java)
            intent1.putExtra("createCode", CREATE_RESULT)
            }
            PLAY_RESULT -> {
                val intent2 = Intent(this, CallGallery::class.java)
                intent2.putExtra("playCode", PLAY_RESULT)
            }
            /*EDIT_RESULT -> {
                val intent3 = Intent(this, CallGallery::class.java)
                intent3.putExtra("editCode", EDIT_RESULT)
            }*/
            SETTINGS_RESULT -> {
                val intent4 = Intent(this, SettingsSystem::class.java)
                intent4.putExtra("settingsCode", SETTINGS_RESULT)
            }
        }
    }

}
